docs/core_docs/docs/modules/data_connection/retrievers/custom.mdx

```diff
@@ -0,0 +1,73 @@
+---
+hide_table_of_contents: true
+sidebar_position: 0
+---
+
+# Custom retrievers
+
+To create your own retriever, you need to implement a `_getRelevantDocuments` method that takes a `string` as its first parameter and an optional `runManager` for tracing.
```

Comment:
I think it could be nice to address the following question:

1) A quick description of what a retriever doc (i.e., given a query and some argument it returns a list of documents)
2) Why should a user implement a BaseRetriever vs. a RunnableLambda that returns a list of Documents. What does inherting from BaseRetriever provide?
3) How does one use runManager? How does one propagate callbacks to any child components of the retriever?